modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayMultiSelect from '@clayui/multi-select';
+import classNames from 'classnames';
+import {fetch} from 'frontend-js-web';
+import React, {useContext, useEffect, useState} from 'react';
+
+import {addToCart} from '../add_to_cart/data';
+import MiniCartContext from './MiniCartContext';
+
+const CHANNEL_RESOURCE_ENDPOINT =
+	'/o/headless-commerce-delivery-catalog/v1.0/channels';
+
+const formattedProducts = [];
```

Comment:
Hmm, this seems odd to be outside the scope of the component, especially because you are pushing to that array. I think this should be component state. Otherwise, you have the risk of mounting/unmounting where you have multiple fetches to the data and you will add the same items to this list. Is that intentional?

Reply:
👍  good call